doc/release/1.11.0-notes.rst

```diff
@@ -174,6 +174,12 @@ This behaviour mimics that of other functions such as ``np.diagonal`` and
 ensures, e.g., that for masked arrays ``np.trace(ma)`` and ``ma.trace()`` give
 the same result.
 
+*np.dot* raises ``TypeError`` instead of ``ValueError``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This behaviour mimics that of other functions such as ``np.inner``. If the two
+arguments cannot be cast to a common type, ``np.dot`` will now always raise a
+``TypeError``. Before it had raise ``ValueError`` in some cases.
```

Comment:
I realized after rereading this there were many issues and so rewrote it. Also, is this in the right place? I wasn't sure where this sort of change went so just stuck it here.